internal/pkg/pkg.go

```diff
@@ -695,3 +699,49 @@ func SetPkgPathAnnotation(rn *yaml.RNode, pkgPath types.UniquePath) error {
 func RemovePkgPathAnnotation(rn *yaml.RNode) error {
 	return rn.PipeE(yaml.ClearAnnotation(pkgPathAnnotation))
 }
+
+// RGFile returns the resourcegroup object by lazy loading it from the filesytem.
+func (p *Pkg) RGFile(filename string) (*rgfilev1alpha1.ResourceGroup, error) {
+	if p.rgFile == nil {
+		// TODO(rquitales): Handle real reading errors vs file does not exist.
+		rg, _ := ReadRGFile(p.UniquePath.String(), filename)
+		p.rgFile = rg
+	}
+	return p.rgFile, nil
+}
+
+// TODO(rquitales): Create new error types for resourcegroup.
+func ReadRGFile(p, filename string) (*rgfilev1alpha1.ResourceGroup, error) {
+	f, err := os.Open(filepath.Join(p, filename))
+	if err != nil {
+		return nil, &KptfileError{
+			Path: types.UniquePath(p),
+			Err:  err,
+		}
+	}
+	defer f.Close()
+
+	rg, err := DecodeRGFile(f)
+	if err != nil {
+		return nil, &KptfileError{
+			Path: types.UniquePath(p),
+			Err:  err,
+		}
+	}
+	return rg, nil
+}
+
+func DecodeRGFile(in io.Reader) (*rgfilev1alpha1.ResourceGroup, error) {
```

Comment:
does this need to be exported ?

Reply:
Function has been merged with ReadRGFile()/